src/StatsMsg.chpl

```diff
@@ -119,17 +119,32 @@ module StatsMsg {
     }
 
     @arkouda.registerCommand()
-    proc cumSum(const ref x: [?d] ?t, axis: int, includeInitial: bool): [] t throws {
+    proc cumSum(const ref x: [?d] ?t, axis: int, includeInitial: bool): [] t throws
+//  proc cumSum(x: [?d] ?t, axis: int, includeInitial: bool): [] t throws
+        where t!= bool { // bool case was already converted to int python-side
+
       if d.rank == 1 {
-        var cs = makeDistArray(if includeInitial then x.size+1 else x.size, t);
-        cs[if includeInitial then 1.. else 0..] = (+ scan x):t;
-        return cs;
+
+//    rewriting some code to hopefully produce a speed improvement
+
+          if !includeInitial {
+            return (+ scan x); 
+          } else {
+            var cs = makeDistArray(x.size+1, t);
+            cs[1..] = (+ scan x);
+            return cs;
+          }
+
+//      var cs = makeDistArray(if includeInitial then x.size+1 else x.size, t);
+//      cs[if includeInitial then 1.. else 0..] = (+ scan x):t;
+//      return cs;
+
       } else {
         var cs = makeDistArray(if includeInitial then expandedDomain(d, axis) else d, t);
 
         forall (slice, _) in axisSlices(d, new list([axis])) {
           const xSlice = removeDegenRanks(x[slice], 1),
-                csSlice = (+ scan xSlice):t;
+                csSlice = (+ scan xSlice); //:t;
```

Comment:
Do you remember why this cast was necessary to begin with? It will be an element-wise operation and could take some time.